protocol/packages/dex/src/swap.rs

```diff
@@ -21,9 +23,9 @@ pub trait ExactAmountIn {
         GIn: Group,
         GSwap: Group;
 
-    fn parse_response<I>(trx_resps: &mut I) -> Result<Amount>
+    fn parse_response<I>(trx_resps: &mut Fuse<I>) -> Result<Amount>
```

Comment:
too much
let's not pollute the interface with an extra requirements on the Iter impl